### PR TITLE
Add support for HiDPI displays in gravatar service

### PR DESCRIPTION
**Pull request by janten**
Please see https://gitlab.com/gitlab-org/gitlab-ce/issues/2781 for details on the issue being fixed here.

### CODE REVIEW

**Comment by janten** [comment]
The test seems to have failed for an unrelated reason. Is it possible to run it again?

**Comment by stanhu** [comment]
The build is failing due an avatar test issue:

```
1) ApplicationHelper avatar_icon should call gravatar_icon when no User exists with the given email
     Failure/Error: helper.avatar_icon('foo@example.com', 20)
       #<#<Class:0x007f5865181810>:0x007f585d2fade8> received :gravatar_icon with unexpected arguments
         expected: ("foo@example.com", 20)
              got: ("foo@example.com", 20, 2)
     # ./app/helpers/application_helper.rb:77:in `avatar_icon'
     # ./spec/helpers/application_helper_spec.rb:100:in `block (3 levels) in <top (required)>'
```

**Comment by janten** [comment]
Should work correctly now.

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -78,12 +78,12 @@ def avatar_icon(user_or_email = nil, size = nil)
     if user
       user.avatar_url(size) || default_avatar
     else
-      gravatar_icon(user_or_email, size)
+      gravatar_icon(user_or_email, size, scales)
```
Comment (stanhu) [inline app/helpers/application_helper.rb]: Did you mean to write `scale`?

**Comment by janten** [comment]
Any news?

**Comment by stanhu** [comment]
Sorry for the delay. Looks good to me. Thanks!

**Pull request merged**